Tests fail against hypothesis 6.39.0

With `pytest-asyncio==0.18.1` and `hypothesis==6.39.0` we are seeing the following test failures. 

Tests were working fine with `hypothesis==6.35.0`, `hypothesis==6.37.2` and `hypothesis==6.38.0`. So tests are starting to fail with 6.39.0

```
=================================== FAILURES ===================================
_______________________________ test_mark_inner ________________________________
tests/hypothesis/test_base.py:19: in test_mark_inner
    @pytest.mark.asyncio
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:470: in run
    self._run()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:872: in _run
    self.generate_new_examples()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:604: in generate_new_examples
    zero_data = self.cached_test_function(bytes(BUFFER_SIZE))
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:1051: in cached_test_function
    self.test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:208: in test_function
    self.__stoppable_test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:184: in __stoppable_test_function
    self._test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:725: in _execute_once_for_engine
    escalate_hypothesis_internal_error()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:699: in _execute_once_for_engine
    result = self.execute_once(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:637: in execute_once
    result = self.test_runner(data, run)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/executors.py:47: in default_new_style_executor
    return function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:633: in run
    return test(*args, **kwargs)
pytest_asyncio/plugin.py:432: in test_mark_inner
    def inner(**kwargs):
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:575: in test
    result = self.test(*args, **kwargs)
E   TypeError: inner() takes 0 positional arguments but 1 was given
---------------------------------- Hypothesis ----------------------------------
You can add @seed(111411370608096322521541188290288019729) to this test or run pytest with --hypothesis-seed=111411370608096322521541188290288019729 to reproduce this failure.
_______________________________ test_mark_outer ________________________________
tests/hypothesis/test_base.py:25: in test_mark_outer
    @given(st.integers())
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:470: in run
    self._run()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:872: in _run
    self.generate_new_examples()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:604: in generate_new_examples
    zero_data = self.cached_test_function(bytes(BUFFER_SIZE))
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:1051: in cached_test_function
    self.test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:208: in test_function
    self.__stoppable_test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:184: in __stoppable_test_function
    self._test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:725: in _execute_once_for_engine
    escalate_hypothesis_internal_error()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:699: in _execute_once_for_engine
    result = self.execute_once(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:637: in execute_once
    result = self.test_runner(data, run)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/executors.py:47: in default_new_style_executor
    return function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:633: in run
    return test(*args, **kwargs)
pytest_asyncio/plugin.py:432: in test_mark_outer
    def inner(**kwargs):
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:575: in test
    result = self.test(*args, **kwargs)
E   TypeError: inner() takes 0 positional arguments but 1 was given
---------------------------------- Hypothesis ----------------------------------
You can add @seed(205756024528203608188762276895564691856) to this test or run pytest with --hypothesis-seed=205756024528203608188762276895564691856 to reproduce this failure.
_________________________ test_mark_and_parametrize[1] _________________________
tests/hypothesis/test_base.py:31: in test_mark_and_parametrize
    @given(x=st.none())
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:470: in run
    self._run()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:872: in _run
    self.generate_new_examples()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:604: in generate_new_examples
    zero_data = self.cached_test_function(bytes(BUFFER_SIZE))
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:1051: in cached_test_function
    self.test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:208: in test_function
    self.__stoppable_test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:184: in __stoppable_test_function
    self._test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:725: in _execute_once_for_engine
    escalate_hypothesis_internal_error()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:699: in _execute_once_for_engine
    result = self.execute_once(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:637: in execute_once
    result = self.test_runner(data, run)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/executors.py:47: in default_new_style_executor
    return function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:633: in run
    return test(*args, **kwargs)
pytest_asyncio/plugin.py:432: in test_mark_and_parametrize
    def inner(**kwargs):
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:575: in test
    result = self.test(*args, **kwargs)
E   TypeError: inner() takes 0 positional arguments but 2 were given
---------------------------------- Hypothesis ----------------------------------
You can add @seed(205296931848593918963767617251055649187) to this test or run pytest with --hypothesis-seed=205296931848593918963767617251055649187 to reproduce this failure.
_________________________ test_mark_and_parametrize[2] _________________________
tests/hypothesis/test_base.py:31: in test_mark_and_parametrize
    @given(x=st.none())
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:470: in run
    self._run()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:870: in _run
    self.reuse_existing_examples()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:533: in reuse_existing_examples
    data = self.cached_test_function(existing)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:1051: in cached_test_function
    self.test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:208: in test_function
    self.__stoppable_test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:184: in __stoppable_test_function
    self._test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:725: in _execute_once_for_engine
    escalate_hypothesis_internal_error()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:699: in _execute_once_for_engine
    result = self.execute_once(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:637: in execute_once
    result = self.test_runner(data, run)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/executors.py:47: in default_new_style_executor
    return function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:633: in run
    return test(*args, **kwargs)
pytest_asyncio/plugin.py:432: in test_mark_and_parametrize
    def inner(**kwargs):
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:575: in test
    result = self.test(*args, **kwargs)
E   TypeError: inner() takes 0 positional arguments but 2 were given
---------------------------------- Hypothesis ----------------------------------
You can add @seed(144345173632399901148817170900088580288) to this test or run pytest with --hypothesis-seed=144345173632399901148817170900088580288 to reproduce this failure.
___________________ test_can_use_fixture_provided_event_loop ___________________
tests/hypothesis/test_base.py:39: in test_can_use_fixture_provided_event_loop
    @pytest.mark.asyncio
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:470: in run
    self._run()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:872: in _run
    self.generate_new_examples()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:604: in generate_new_examples
    zero_data = self.cached_test_function(bytes(BUFFER_SIZE))
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:1051: in cached_test_function
    self.test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:208: in test_function
    self.__stoppable_test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:184: in __stoppable_test_function
    self._test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:725: in _execute_once_for_engine
    escalate_hypothesis_internal_error()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:699: in _execute_once_for_engine
    result = self.execute_once(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:637: in execute_once
    result = self.test_runner(data, run)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/executors.py:47: in default_new_style_executor
    return function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:633: in run
    return test(*args, **kwargs)
pytest_asyncio/plugin.py:432: in test_can_use_fixture_provided_event_loop
    def inner(**kwargs):
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:575: in test
    result = self.test(*args, **kwargs)
E   TypeError: inner() takes 0 positional arguments but 2 were given
---------------------------------- Hypothesis ----------------------------------
You can add @seed(32806281167570098611214284431267867570) to this test or run pytest with --hypothesis-seed=32806281167570098611214284431267867570 to reproduce this failure.
____________________________ test_async_auto_marked ____________________________
/build/source/tests/hypothesis/test_base.py:64: in test_async_auto_marked
    result.assert_outcomes(passed=1)
E   AssertionError: assert {'errors': 0,...pped': 0, ...} == {'errors': 0,...pped': 0, ...}
E     Omitting 4 identical items, use -vv to show
E     Differing items:
E     {'passed': 0} != {'passed': 1}
E     {'failed': 1} != {'failed': 0}
E     Use -v to get the full diff
----------------------------- Captured stdout call -----------------------------
============================= test session starts ==============================
platform linux -- Python 3.9.10, pytest-7.0.1, pluggy-1.0.0
rootdir: /build/pytest-of-nixbld/pytest-0/test_async_auto_marked0
plugins: asyncio-0.18.1, flaky-3.7.0, hypothesis-6.39.0
asyncio: mode=auto
collected 1 item

test_async_auto_marked.py F                                              [100%]

=================================== FAILURES ===================================
_______________________________ test_hypothesis ________________________________

    @given(n=st.integers())
>   async def test_hypothesis(n: int):

test_async_auto_marked.py:9:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:470: in run
    self._run()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:872: in _run
    self.generate_new_examples()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:604: in generate_new_examples
    zero_data = self.cached_test_function(bytes(BUFFER_SIZE))
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:1051: in cached_test_function
    self.test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:208: in test_function
    self.__stoppable_test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:184: in __stoppable_test_function
    self._test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:725: in _execute_once_for_engine
    escalate_hypothesis_internal_error()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:699: in _execute_once_for_engine
    result = self.execute_once(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:637: in execute_once
    result = self.test_runner(data, run)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/executors.py:47: in default_new_style_executor
    return function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:633: in run
    return test(*args, **kwargs)
/build/source/pytest_asyncio/plugin.py:432: in test_hypothesis
    def inner(**kwargs):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (0,), kwargs = {}, initial_draws = 1, start = 8032879.46620021

    @proxies(self.test)
    def test(*args, **kwargs):
        self.__test_runtime = None
        initial_draws = len(data.draw_times)
        start = time.perf_counter()
>       result = self.test(*args, **kwargs)
E       TypeError: inner() takes 0 positional arguments but 1 was given

/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:575: TypeError
---------------------------------- Hypothesis ----------------------------------
You can add @seed(5058031335671856716915173869561083396) to this test or run pytest with --hypothesis-seed=5058031335671856716915173869561083396 to reproduce this failure.
=========================== short test summary info ============================
FAILED test_async_auto_marked.py::test_hypothesis - TypeError: inner() takes ...
============================== 1 failed in 0.18s ===============================
___________________________ TestOne.test_hypothesis ____________________________
tests/hypothesis/test_inherited_test.py:8: in test_hypothesis
    @given(value=st.integers())
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:470: in run
    self._run()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:872: in _run
    self.generate_new_examples()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:604: in generate_new_examples
    zero_data = self.cached_test_function(bytes(BUFFER_SIZE))
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:1051: in cached_test_function
    self.test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:208: in test_function
    self.__stoppable_test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:184: in __stoppable_test_function
    self._test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:725: in _execute_once_for_engine
    escalate_hypothesis_internal_error()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:699: in _execute_once_for_engine
    result = self.execute_once(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:637: in execute_once
    result = self.test_runner(data, run)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/executors.py:47: in default_new_style_executor
    return function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:633: in run
    return test(*args, **kwargs)
pytest_asyncio/plugin.py:432: in test_hypothesis
    def inner(**kwargs):
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:575: in test
    result = self.test(*args, **kwargs)
E   TypeError: inner() takes 0 positional arguments but 2 were given
---------------------------------- Hypothesis ----------------------------------
You can add @seed(220747386016761914034003099428413956254) to this test or run pytest with --hypothesis-seed=220747386016761914034003099428413956254 to reproduce this failure.
___________________________ TestTwo.test_hypothesis ____________________________
tests/hypothesis/test_inherited_test.py:8: in test_hypothesis
    @given(value=st.integers())
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:470: in run
    self._run()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:870: in _run
    self.reuse_existing_examples()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:533: in reuse_existing_examples
    data = self.cached_test_function(existing)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:1051: in cached_test_function
    self.test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:208: in test_function
    self.__stoppable_test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/internal/conjecture/engine.py:184: in __stoppable_test_function
    self._test_function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:725: in _execute_once_for_engine
    escalate_hypothesis_internal_error()
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:699: in _execute_once_for_engine
    result = self.execute_once(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:637: in execute_once
    result = self.test_runner(data, run)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/executors.py:47: in default_new_style_executor
    return function(data)
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:633: in run
    return test(*args, **kwargs)
pytest_asyncio/plugin.py:432: in test_hypothesis
    def inner(**kwargs):
/nix/store/lrq051800ykl7b8ddqlm40p0l2mc75m1-python3.9-hypothesis-6.39.0/lib/python3.9/site-packages/hypothesis/core.py:575: in test
    result = self.test(*args, **kwargs)
E   TypeError: inner() takes 0 positional arguments but 2 were given
---------------------------------- Hypothesis ----------------------------------
You can add @seed(283343356983479488199690035036288948312) to this test or run pytest with --hypothesis-seed=283343356983479488199690035036288948312 to reproduce this failure.
=========================== short test summary info ============================
SKIPPED [1] tests/test_simple.py:238: Test a skip error inside asyncio
XFAIL tests/test_simple.py::test_asyncio_marker_fail
  need a failure
============== 8 failed, 64 passed, 1 skipped, 1 xfailed in 7.31s ==============
```

They all seem to boil down to this line:

> E   TypeError: inner() takes 0 positional arguments but 2 were given

Happy to test patches.

--- a/README.rst
+++ b/README.rst
@@ -260,6 +260,7 @@ Changelog
 0.18.2 (Unreleased)
 ~~~~~~~~~~~~~~~~~~~
 - Fix asyncio auto mode not marking static methods. `#295 <https://github.com/pytest-dev/pytest-asyncio/issues/295>`_
+- Fix a compatibility issue with Hypothesis 6.39.0. `#302 <https://github.com/pytest-dev/pytest-asyncio/issues/302>`_
 
 
 0.18.1 (22-02-10)

--- a/pytest_asyncio/plugin.py
+++ b/pytest_asyncio/plugin.py
@@ -432,8 +432,8 @@ def wrap_in_sync(
         func = raw_func
 
     @functools.wraps(func)
-    def inner(**kwargs):
-        coro = func(**kwargs)
+    def inner(*args, **kwargs):
+        coro = func(*args, **kwargs)
         if not inspect.isawaitable(coro):
             pyfuncitem.warn(
                 pytest.PytestWarning(

